libsql-wal/src/segment/current.rs

```diff
@@ -350,29 +350,36 @@ impl<F> CurrentSegment<F> {
         &self.tail
     }
 
-    // todo: maybe return boxed frames?
-    pub fn rev_frame_stream(&self) -> impl Stream<Item = Result<Frame>> + '_
+    /// returns all the frames that changed between start_frame_no and the current commit index
+    pub(crate) fn frame_stream_from(
+        &self,
+        start_frame_no: u64,
+    ) -> impl Stream<Item = Result<Frame>> + '_
     where
         F: FileExt,
     {
-        async_stream::try_stream! {
-            let (start_frame_no, last_committed) = {
-                let header = self.header.lock();
-                (header.start_frame_no.get(), header.last_commited_frame_no.get())
-            };
-            let mut next_offset = (last_committed - start_frame_no) as u32;
-            loop {
-                let byte_offset = frame_offset(next_offset);
+        let (last_committed, size_after) = {
+            let header = self.header.lock();
+            (header.last_committed(), header.db_size.get())
+        };
 
+        // we get all frame offsets of pages that changed since start_frame_no
+        let mut iter = self.index.iter(start_frame_no, last_committed).peekable();
+
+        async_stream::try_stream! {
```

Comment:
consider using tokio's consume budget to make sure that we yield in a friendly manner?

Reply:
Can you explain why we would need that?